exporter/kineticaexporter/config.go

```diff
@@ -0,0 +1,253 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package kineticaexporter // import
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configopaque"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/natefinch/lumberjack.v2"
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr = "kinetica"
+
+	maxSize              = 10
+	maxBackups           = 5
+	maxAge               = 10
+	defaultLogConfigFile = "config_log_zap.yaml"
+)
+
+var (
+	kineticaLogger *zap.Logger
+)
+
+// Config defines configuration for the Kinetica exporter.
+type Config struct {
+	Host               string              `mapstructure:"host"`
+	Schema             string              `mapstructure:"schema"`
+	Username           string              `mapstructure:"username"`
+	Password           configopaque.String `mapstructure:"password"`
+	BypassSslCertCheck bool                `mapstructure:"bypasssslcertcheck"`
+	LogConfigFile      string              `mapstructure:"logconfigfile"`
+}
+
+// Validate the config
+//
+//	@receiver cfg
+//	@return error
+func (cfg *Config) Validate() error {
+	kineticaHost, err := url.ParseRequestURI(cfg.Host)
+	if err != nil {
+		return err
+	}
+	if kineticaHost.Scheme != "http" && kineticaHost.Scheme != "https" {
+		return errors.New("Protocol must be either `http` or `https`")
+	}
+
+	if !fileExists(cfg.LogConfigFile) {
+		fmt.Println("WARNING : LOG config file ", cfg.LogConfigFile, " does not exist; will work with default logger ...")
+	}
+
+	fmt.Println("Password = ", string(cfg.Password))
+
+	return nil
+}
+
+func parseNumber(s string, fallback int) int {
+	v, err := strconv.Atoi(s)
+	if err == nil {
+		return v
+	}
+	return fallback
+}
+
+// createLogger - creates a Logger using user defined configuration from a file.
+// If the file name is not found or there is some error it will create a default
+// logger using the default logger config file named 'config_log_zap.yaml'.
+//
+//	@receiver cfg
+//	@return *zap.Logger
+func (cfg *Config) createLogger() *zap.Logger {
```

Comment:
Is this part of the exporting logic? If it is creating a logger to use within the component but not as part of the data exporting then we should be using the logger that the collector provides instead.  If it is exporting logic please remove it from this PR and we'll add it in a future PR.

Reply:
Removed this method since this was intended to be specific to our internal logging requirements.